nodebuilder/share/module.go

```diff
@@ -241,38 +206,34 @@ func edsStoreComponents(cfg *Config) fx.Option {
 	)
 }
 
-func fullAvailabilityComponents() fx.Option {
-	return fx.Options(
-		fx.Provide(fx.Annotate(
-			full.NewShareAvailability,
-			fx.OnStart(func(ctx context.Context, avail *full.ShareAvailability) error {
-				return avail.Start(ctx)
-			}),
-			fx.OnStop(func(ctx context.Context, avail *full.ShareAvailability) error {
-				return avail.Stop(ctx)
+func availabilityComponents(tp node.Type, cfg *Config) fx.Option {
+	switch tp {
+	case node.Light:
+		return fx.Options(
+			fx.Provide(fx.Annotate(
+				func(getter share.Getter, ds datastore.Batching) *light.ShareAvailability {
+					return light.NewShareAvailability(
+						getter,
+						ds,
+						light.WithSampleAmount(cfg.LightAvailability.SampleAmount),
+					)
+				},
+				fx.OnStop(func(ctx context.Context, la *light.ShareAvailability) error {
+					return la.Close(ctx)
+				}),
+			)),
+			fx.Provide(func(avail *light.ShareAvailability) share.Availability {
+				return avail
 			}),
-		)),
-		fx.Provide(func(avail *full.ShareAvailability) share.Availability {
-			return avail
-		}),
-	)
-}
-
-func lightAvailabilityComponents(cfg *Config) fx.Option {
-	return fx.Options(
-		fx.Provide(fx.Annotate(
-			light.NewShareAvailability,
-			fx.OnStop(func(ctx context.Context, la *light.ShareAvailability) error {
-				return la.Close(ctx)
+		)
+	case node.Bridge, node.Full:
+		return fx.Options(
+			fx.Provide(full.NewShareAvailability),
+			fx.Provide(func(avail *full.ShareAvailability) share.Availability {
+				return avail
 			}),
-		)),
-		fx.Provide(func() []light.Option {
-			return []light.Option{
-				light.WithSampleAmount(cfg.LightAvailability.SampleAmount),
-			}
-		}),
-		fx.Provide(func(avail *light.ShareAvailability) share.Availability {
-			return avail
-		}),
-	)
+		)
+	default:
+		panic("invalid node type")
```

Comment:
Consider returning error like "Node type: %s is not supported for availability"

Reply:
we do this everywhere else.